botcraft/src/AI/Tasks/InventoryTasks.cpp

```diff
@@ -359,12 +359,12 @@ namespace Botcraft
                 std::sort( // Find the face face closest to player looking direction
                     face_candidates.begin(), face_candidates.end(), [&](PlayerDiggingFace a, PlayerDiggingFace b) -> bool
                     {
-                        Vector3<double> a_offset =  neighbour_offsets[static_cast<int>(a)];
+                        Vector3<double> a_offset = neighbour_offsets[static_cast<int>(a)];
                         Vector3<double> b_offset = neighbour_offsets[static_cast<int>(b)];
-                        Vector3<double> a_to_player = a_offset - pos + Vector3<double>(0.5); 
-                        Vector3<double> b_to_player = b_offset - pos + Vector3<double>(0.5);
-                        return player_orientation.dot(a_to_player) > player_orientation.dot(b_to_player); });
-                face = face_candidates.front();
+                        return player_orientation.dot(a_offset) < player_orientation.dot(b_offset); // a < b because a negative dot product means the vectors are in opposite directions IE the player is looking at the face.
```

Comment:
I don't think it's working. Now it doesn't depend on the position of the block anymore, which means that it will return the same face for any block. The drawing below shows a case where it should return top face (i.e. bottom face of the neighbour) for A, but right face (i.e. left face of the neighbour block) for block B.

![image](https://github.com/adepierre/Botcraft/assets/24371370/0ecd086e-625e-4744-9df7-41145413d41c)